src/driver/amdxdna/aie2_event_trace.c

```diff
@@ -0,0 +1,329 @@
+		ret = -ENOMEM;
+		goto free_dma_trace_buf;
+	}
+
+	req_buf->log_ch_irq = pci_irq_vector(to_pci_dev(xdna->ddev.dev), msi_idx);
+	ret = request_irq(req_buf->log_ch_irq, log_buffer_irq_handler, 0, "LOG_BUFFER", ndev);
+	if (ret) {
+		XDNA_ERR(xdna, "Failed to register irq %d ret %d", msi_idx, ret);
+		goto destroy_wq;
+	}
+
+	req_buf->fw_log_buf = kzalloc((sizeof(u8) * TRACE_EVENT_BUFFER_SIZE), GFP_KERNEL);
+	if (!req_buf->fw_log_buf) {
+		XDNA_ERR(xdna, "Failed to allocate kernel log buffer");
+		ret = -ENOMEM;
+		goto destroy_wq;
+	}
+	return 0;
+
+destroy_wq:
+	destroy_workqueue(req_buf->wq);
+free_dma_trace_buf:
+	dma_free_noncoherent(xdna->ddev.dev, req_buf->trace_req.dram_buffer_size, req_buf->buf,
+		(dma_addr_t)req_buf->trace_req.dram_buffer_address, DMA_BIDIRECTIONAL);
+	kfree(req_buf);
+	ndev->event_trace_req = NULL;
+	return ret;
+}
+
+void aie2_set_trace_timestamp(struct amdxdna_dev_hdl *ndev, u32 *resp)
+{
+	struct start_event_trace_resp *trace_resp = (struct start_event_trace_resp *)resp;
+
+	ndev->event_trace_req->resp_timestamp = trace_resp->current_timestamp;
+	ndev->event_trace_req->sys_start_time = ktime_get_ns() / 1000; /*Convert ns to us*/
+	aie2_register_log_buf_irq_hdl(ndev, trace_resp->msi_idx);
+}
+
+static int aie2_stop_event_trace_send(struct amdxdna_dev_hdl *ndev)
+{
+	struct amdxdna_dev *xdna = ndev->xdna;
+
+	if (!ndev->event_trace_req) {
+		XDNA_DBG(xdna, "Event tracing is not started");
+		return 0;
+	}
+
+	drm_WARN_ON(&xdna->ddev, !mutex_is_locked(&xdna->dev_lock));
+	return aie2_stop_event_trace(ndev);
+}
+
+static int aie2_event_trace_alloc(struct amdxdna_dev_hdl *ndev)
+{
+	struct amdxdna_dev *xdna = ndev->xdna;
+	struct event_trace_req_buf *req_buf = ndev->event_trace_req;
+
+	req_buf->buf = dma_alloc_noncoherent(xdna->ddev.dev, TRACE_EVENT_BUFFER_SIZE,
+					(dma_addr_t *)&req_buf->trace_req.dram_buffer_address, DMA_BIDIRECTIONAL, GFP_KERNEL);
+
+	if (!req_buf->buf)
+		return -ENOMEM;
+
+	req_buf->trace_req.dram_buffer_size = TRACE_EVENT_BUFFER_SIZE;
+	XDNA_DBG(ndev->xdna, "Start event trace buf addr: 0x%llx size 0x%x", req_buf->trace_req.dram_buffer_address,
+			 req_buf->trace_req.dram_buffer_size);
+	return 0;
+}
+
+static void aie2_event_trace_free(struct amdxdna_dev_hdl *ndev)
+{
+	struct amdxdna_dev *xdna = ndev->xdna;
+	struct event_trace_req_buf *req_buf = ndev->event_trace_req;
+
+	if (!req_buf)
+		return;
+
+	cancel_work_sync(&req_buf->work);
+	destroy_workqueue(req_buf->wq);
+	dma_free_noncoherent(xdna->ddev.dev, req_buf->trace_req.dram_buffer_size, req_buf->buf,
+			(dma_addr_t)req_buf->trace_req.dram_buffer_address, DMA_BIDIRECTIONAL);
+
+	kfree(ndev->event_trace_req->fw_log_buf);
+	free_irq(req_buf->log_ch_irq, ndev);
+	kfree(req_buf);
+	ndev->event_trace_req = NULL;
+}
+
+static int aie2_start_event_trace_send(struct amdxdna_dev_hdl *ndev)
+{
+	int ret;
+	struct amdxdna_dev *xdna = ndev->xdna;
+	struct event_trace_req_buf *trace_req_buf = ndev->event_trace_req;
```

Comment:
Linux coding style require reverse XMAS tree declarations. Same for other functions.
```suggestion
	struct event_trace_req_buf *trace_req_buf = ndev->event_trace_req;
	struct amdxdna_dev *xdna = ndev->xdna;
	int ret;
```